### PR TITLE
feat: add KubectLayer and release as @aws-cdk/lambda-layer-kubectl-v25

### LINKED ISSUE #166: Feature Request: Asset with kubectl v1.25
Kubernetes v1.25 [is now supported by EKS](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html#kubernetes-1.25).

Edit: I'm willing to contribute to this feature

### PR DESCRIPTION
**Pull request by CarlosDomingues**
Adds support for kubectl 1.25

Closes #166.

### CODE REVIEW

File: API.md.md [hunk 1]
```diff
@@ -2,32 +2,32 @@
 
 ## Constructs <a name="Constructs" id="Constructs"></a>
 
-### KubectlV24Layer <a name="KubectlV24Layer" id="@aws-cdk/lambda-layer-kubectl-v24.KubectlV24Layer"></a>
+### Kubectlv25Layer <a name="Kubectlv25Layer" id="@aws-cdk/lambda-layer-kubectl-v25.Kubectlv25Layer"></a>
```
Comment (kaizencc) [inline API.md.md]: lmaoooo please delete this file. I've added the same change to `kubectl-v24/main`. Was a bug in jsii-docgen that's been fixed [here](https://github.com/cdklabs/jsii-docgen/pull/934/files).

**Review by kaizencc — COMMENTED** [review]
This all looks good with just one minor comment (that's our fault lol). Can I trust that you've actually deployed the integ tests and not just updated the snapshots?

**Comment by CarlosDomingues** [comment]
> This all looks good with just one minor comment (that's our fault lol)

Ok, will fix that later today.

EDIT: Just realized you made a PR earlier 😅

> Can I trust that you've actually deployed the integ tests and not just updated the snapshots?

Yes. Please check the screenshots attached and let me know whether the tests looks correct.

deploy-01

deploy-02

**Comment by kaizencc** [comment]
Thank you for the screenshots! They look perfect. I'm good with this PR, and once it's released, we can reference it in the CDK repo as an option. Edit: see ensuing comment.

File: README.md [hunk 2]
```diff
@@ -9,23 +9,23 @@
...
 
-This module exports a single class called `KubectlV24Layer` which is a `lambda.LayerVersion` that
+This module exports a single class called `KubectlV25Layer` which is a `lambda.LayerVersion` that
 bundles the [`kubectl`](https://kubernetes.io/docs/reference/kubectl/kubectl/) and the
 [`helm`](https://helm.sh/) command line.
 
-> - Helm Version: 3.9.4
-> - Kubectl Version: 1.24.0
-> 
+> - Helm Version: 3.10.3
```
Comment (kaizencc) [inline README.md]: Shoot, i think we want Helm version 3.11.1, that supports kubectl v1.25 too, right? @CarlosDomingues

Comment (CarlosDomingues) [reply README.md]: Checking [here](https://helm.sh/docs/topics/version_skew/#supported-version-skew) again, looks like I missed Helm 3.11.x. Oops. Will fix that in a bit.

Comment (CarlosDomingues) [reply README.md]: I've updated helm support to 3.11.1. Please find all tests attached:

deploy-01

deploy-04

File: src/kubectl-layer.ts [hunk 3]
```diff
@@ -5,13 +5,13 @@ import { assetHash, ASSET_FILE } from './_asset';
...
  */
-export class KubectlV24Layer extends lambda.LayerVersion {
+export class KubectlV25Layer extends lambda.LayerVersion {
   constructor(scope: Construct, id: string) {
     super(scope, id, {
       code: lambda.Code.fromAsset(ASSET_FILE, {
         assetHash: assetHash(),
       }),
-      description: '/opt/kubectl/kubectl 1.24; /opt/helm/helm 3.9',
+      description: '/opt/kubectl/kubectl 1.25; /opt/helm/helm 3.10',
```
Comment (kaizencc) [inline src/kubectl-layer.ts]:
```suggestion
      description: '/opt/kubectl/kubectl 1.25; /opt/helm/helm 3.11',
```

File: test/kubectl-layer.test.ts [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
   const stack = new Stack();
 
   // WHEN
-  new KubectlV24Layer(stack, 'MyLayer');
+  new KubectlV25Layer(stack, 'MyLayer');
 
   // THEN
   Template.fromStack(stack).hasResourceProperties('AWS::Lambda::LayerVersion', {
-    Description: '/opt/kubectl/kubectl 1.24; /opt/helm/helm 3.9',
+    Description: '/opt/kubectl/kubectl 1.25; /opt/helm/helm 3.10',
```
Comment (kaizencc) [inline test/kubectl-layer.test.ts]:
```suggestion
    Description: '/opt/kubectl/kubectl 1.25; /opt/helm/helm 3.11',
```

File: test/kubectl-layer.integ.snapshot/lambda-layer-kubectl-integ-stack.template.json [hunk 5]
```diff
@@ -7,9 +7,9 @@
           "S3Bucket": {
             "Fn::Sub": "cdk-hnb659fds-assets-${AWS::AccountId}-${AWS::Region}"
           },
-          "S3Key": "c475180f5b1bbabac165414da13a9b843b111cd3b6d5fae9c954c006640c4064.zip"
+          "S3Key": "ebc48a9a74e277f463f78c5a0b3bcf489a891efe25ece75b5b439b9e89bee4ac.zip"
         },
-        "Description": "/opt/kubectl/kubectl 1.24; /opt/helm/helm 3.9",
+        "Description": "/opt/kubectl/kubectl 1.25; /opt/helm/helm 3.10",
```
Comment (kaizencc) [inline test/kubectl-layer.integ.snapshot/lambda-layer-kubectl-integ-stack.template.json]:
```suggestion
        "Description": "/opt/kubectl/kubectl 1.25; /opt/helm/helm 3.11",
```

**Comment by kaizencc** [comment]
@CarlosDomingues thanks for your work so far :). We are basically done but as you can see, the lambda description was still referencing helm 3.10. I updated the description, but I think we're gonna get unlucky because the description is part of the lambda hash so the snapshot may need updating. Once that's updated, I shall approve!

**Comment by CarlosDomingues** [comment]
Just to confirm, do I need to re-deploy the project with the correct description to update the hashes and commit them?

**Comment by kaizencc** [comment]
@CarlosDomingues nah, just snapshot update is ok in this case :)

**Comment by robertd** [comment]
@CarlosDomingues @kaizencc Are we using helm 3.10.x just for backward support of k8s 1.22? https://helm.sh/docs/topics/version_skew/

**Comment by robertd** [comment]
Scratch my previous comment... I just saw that helm was brought up to 3.11.x which supports k8s 1.23.x - 1.26.x. Carry on :)

**Review by kaizencc — APPROVED** [review]
@CarlosDomingues thanks for your contribution!

**Pull request merged**